### PR TITLE
[SYCL] test for ONEAPI_DEVICE_SELECTOR case sensitivity fix

### PR DESCRIPTION
**Pull request by cperkinsintel**
fix for ONEAPI_DEVICE_SELECTOR case sensitivity is here: https://github.com/intel/llvm/pull/8238

### CODE REVIEW

File: SYCL/OneapiDeviceSelector/case_sensitivity.cpp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+// does not actually require OpenCL or GPU. Just testing parsing.
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %S/Inputs/trivial.cpp -o %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR="OPENCL:*" %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR="opencl:*" %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR="*:GPU" %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR="*:gpu" %t.out
```
Comment (steffenlarsen) [inline SYCL/OneapiDeviceSelector/case_sensitivity.cpp]:
```suggestion
// RUN: env ONEAPI_DEVICE_SELECTOR="*:gpu" %t.out

```

Comment (steffenlarsen) [inline SYCL/OneapiDeviceSelector/case_sensitivity.cpp]: Would it make sense to have a test where it mixes upper and lowercase letters? I.e. something like `OPeNcL` or even an actual common spelling like `OpenCL`.

Comment (cperkinsintel) [reply SYCL/OneapiDeviceSelector/case_sensitivity.cpp]: Maybe. What if I removed all these cases for just one: "OpEnCl:GpU" ?

**Comment by cperkinsintel** [comment]
The "regular" test failures are expected. The Jenkins/llvm-test-suite run is the run against the PR with the new feature, and all tests are passing.

**Review by againull — APPROVED** [review]

**Pull request merged**